config/confighttp/confighttp.go

```diff
@@ -132,11 +138,28 @@ func (hcs *HTTPClientSettings) ToClient(host component.Host, settings component.
 		transport.IdleConnTimeout = *hcs.IdleConnTimeout
 	}
 
-	clientTransport := (http.RoundTripper)(transport)
+	if len(hcs.Headers) > 0 && len(hcs.OpaqueHeaders) > 0 {
+		return nil, fmt.Errorf("fields Headers and OpaqueHeaders were set at the same time, use only OpaqueHeaders")
+	}
+
+	headers := map[string]string{}
+
+	if len(hcs.OpaqueHeaders) > 0 {
+		for k, v := range hcs.OpaqueHeaders {
+			headers[k] = string(v)
+		}
+	}
+
 	if len(hcs.Headers) > 0 {
+		headers = hcs.Headers
+		settings.Logger.Warn("confighttp.HTTPClientSettings.Headers is deprecated in favor of confighttp.HTTPClientSettings.OpaqueHeaders")
+	}
+
+	clientTransport := (http.RoundTripper)(transport)
+	if len(headers) > 0 {
 		clientTransport = &headerRoundTripper{
 			transport: transport,
-			headers:   hcs.Headers,
+			headers:   headers,
```

Comment:
Would it make sense to convert the string headers to opaque headers here?

Reply:
Sure, it doesn't hurt to use the opaque type there too :)